addons/mail/static/src/components/message_reaction_per_emoji/message_reaction_per_emoji.js

```diff
@@ -0,0 +1,85 @@
+/** @odoo-module **/
+
+const { Component } = owl;
+import { registerMessagingComponent } from '@mail/utils/messaging_component';
+
+export class MessageReactionPerEmoji extends Component {
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    get count() {
+        if (this.messageReactionPerEmoji) {
+            return this.messageReactionPerEmoji.count;
+        }
+    }
+
+    get message() {
+        if (this.messageReactionPerEmoji) {
+            return this.messageReactionPerEmoji.message;
+        }
+    }
+
+    get messageReactionPerEmoji() {
+        return this.messaging.models['mail.message_reaction_per_emoji'].get(this.props.messageReactionPerEmojiLocalId);
+    }
+
+    get emojiUnicode() {
+        if (this.messageReactionPerEmoji) {
+            return this.messageReactionPerEmoji.emoji_unicode;
+        }
+    }
+
+    get hasUserReacted() {
+        return this.messageReactionPerEmoji && this.messageReactionPerEmoji.hasUserReacted;
+    }
+
+    get reactionsSummary() {
+        const reactions = this.messaging.models['mail.message_reaction'].all().filter((reaction) =>
+            reaction.message === this.message && reaction.emoji_unicode === this.emojiUnicode);
+        const reactors = reactions.map((reaction) => reaction.reactor);
+        let reactors_text = "";
+        for (let i = 0; i < reactors.length; i++) {
+            if (i === 0) {
+                reactors_text = reactors[i].name + " ";
+            } else if (reactors.length === 2 || i === reactors.length - 1) {
+                reactors_text = reactors_text + "and " + reactors[i].name + " ";
+            } else {
+                reactors_text = reactors_text + ", " + reactors[i].name + " ";
+            }
+        }
+        if (reactors.length === 1) {
+            reactors_text = reactors_text + "has reacted with ";
+        } else {
+            reactors_text = reactors_text + "have reacted with ";
+        }
+        reactors_text = reactors_text + this.emojiUnicode;
+        return reactors_text;
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onClick() {
+        this.messageReactionPerEmoji.removeReaction();
```

Comment:
Prefer dispatching to model while keeping the info on the user interaction. So something like `onClick` method on the `MessageReactionPerEmoji` model.

_(btw something looks off with naming of this model. See suggestions to address this issue in a comment on file defining this model)_